models/activities/notification.go

```diff
@@ -341,6 +345,90 @@ func GetIssueNotification(ctx context.Context, userID, issueID int64) (*Notifica
 	return notification, err
 }
 
+// CreateOrUpdateIssueNotifications creates an release notification
+// for each watcher, or updates it if already exists
+func CreateOrUpdateReleaseNotifications(releaseID, receiverID int64) error {
+	ctx, committer, err := db.TxContext(db.DefaultContext)
```

Comment:
Please don't, just pass a `ctx` param instead (and perhaps wrap that in a transaction).

Reply:
This function is called from a Queue, so there is no ctx to pass.

Reply:
Ah, I hate that.
What about a comment why it is needed ?